### PR TITLE
Enhanced Reference Implementation for EIP-5289

### PR DESCRIPTION
**Pull request by jaerith**
By the way, I love this idea!  I was thinking that it could be integrated with EIP-4361 (i.e., SIWE).  In fact, I have a reference implementation that would be a perfect place to demonstrate such an integration:

https://docs.login.xyz/additional-support/community-highlights (the siwe-nethereum one)

I'd love to hear any possible feedback about this integration in the Issues section!

### CODE REVIEW

**Comment by xinbenlv** [comment]
LGTM

File: assets/eip-5289/interfaces/IERC165.sol [hunk 1]
```diff
@@ -8,5 +8,5 @@ interface IERC165 {
     ///  uses less than 30,000 gas.
     /// @return `true` if the contract implements `interfaceID` and
     ///  `interfaceID` is not 0xffffffff, `false` otherwise
-    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+    function supportsInterface(bytes4 interfaceID) external pure returns (bool);
```
[the remark this answers is not in the document]
Comment (jaerith) [reply assets/eip-5289/interfaces/IERC165.sol]: No problem.  Consider it done!

File: assets/eip-5289/ERC5289Library.sol [hunk 2]
```diff
@@ -1,38 +1,45 @@
...
         return signedAt[documentId][user] != 0;
     }
 
     function documentSignedAt(address user, uint16 documentId) public view returns (uint64 timestamp) {
         return signedAt[documentId][user];
     }
 
-    function signDocument() public {
+    function signDocument(address signer, uint16 documentId, bytes memory signature) public {
+
```
[the remark this answers is not in the document]
Comment (jaerith) [reply assets/eip-5289/ERC5289Library.sol]: No problem.  Consider it done!

File: assets/eip-5289/interfaces/IERC5289Library.sol [hunk 3]
```diff
@@ -1,23 +1,21 @@
 /// SPDX-License-Identifier: CC0-1.0
 pragma solidity ^0.8.0;
 
-import "./IERC165.sol";
-
-interface IERC5289Library is IERC165 {
+interface IERC5289Library {
```
[the remark this answers is not in the document]
Comment (jaerith) [reply assets/eip-5289/interfaces/IERC5289Library.sol]: No problem.  Consider it done!

**Review by Pandapip1 — APPROVED** [review]
@SamWilsn please manually merge.

**Pull request merged**